src/components/LayerTree/LayerTree.js

```diff
@@ -101,13 +101,15 @@ class LayerTree extends Component {
     const { layerService, isItemHidden } = this.props;
     const initialExpandedLayerNames =
       layerService && layerService.getLayers()
-        ? layerService
-            .getLayers()
-            .filter(
-              l =>
-                !isItemHidden(l) &&
-                l.getVisibleChildren().filter(c => !isItemHidden(c)).length,
-            )
+        ? this.getExpandedLayers(
+            layerService
+              .getLayers()
+              .filter(
+                l =>
+                  !isItemHidden(l) &&
+                  l.getVisibleChildren().filter(c => !isItemHidden(c)).length,
+              ),
+          )
```

Comment:
Is  ` this.getExpandedLayers(layerService.getLayers())`  not enough ?

Reply:
In the first level we expand all layers which are not hidden and have visible children that are not hidden.
Beginning with the second level, we expand the layers that are not hidden and have the attribute `isAlwaysExpanded`, no matter if its children are visible or not. If one of these layers is expanded, we look at the next level (recursively).
So, the rules are different for the first level and the following levels.